project-koku/koku
# 1321 ocp markup

#1321. Also discovered that infra_cost was returning as null. This PR addresses both issues.

This bug is related to #1175. 

This PR reverts to using the CostSummary table until the simplification of OCP tables can be nailed down. While looking through the tests, I noticed that the tests written for the costs were summing to zero and passing every time. The helper used to populate the database was not filling tables in the correct order. The helper now fills the CostSummary table last so that the costs will no longer sum to zero.

Testing:
1. Add an ocp provider
2. Upload some data for the provider
3. Add a cost model with rates and markup, wait until the data processes and summarizes
4. See that markup cost of the OCP provider is not zero. Also see that infrastructure_cost is not null.

example results with 10% markup:
```
"total": {
            "cost": {
                "value": 3294.225,
                "units": "USD"
            },
            "infrastructure_cost": {
                "value": 0.0,
                "units": "USD"
            },
            "derived_cost": {
                "value": 2994.75,
                "units": "USD"
            },
            "markup_cost": {
                "value": 299.475,
                "units": "USD"
            }
        }
```
